src/test/text_streamer_test.cpp

```diff
@@ -0,0 +1,150 @@
+    auto tokens = tokenizer->encode("TEST");
+    std::vector<int64_t> expectedTokens = {565, 4923};
+    assertTokensValues(tokens, expectedTokens);
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
+        EXPECT_FALSE(rval.has_value());
+    }
+}
+
+TEST_F(TextStreamerTest, valueReturned) {
+    std::string testPrompt = "TEST\n";
+    auto tokens = tokenizer->encode(testPrompt);
+    std::vector<int64_t> expectedTokens = {565, 4923, 50118};
+    assertTokensValues(tokens, expectedTokens);
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
+        if (i < tokens.get_size() - 1) {  // No value returned until last token with new line passed to tokenizer
+            EXPECT_FALSE(rval.has_value());
+        } else {
+            EXPECT_TRUE(rval.has_value());
+            EXPECT_EQ(rval.value().compare(testPrompt), 0);
+        }
+    }
+}
+
+TEST_F(TextStreamerTest, valueReturnedCacheNotCleared) {
+    std::string testPrompt1 = "TEST";
+    auto tokens = tokenizer->encode(testPrompt1);
+    std::vector<int64_t> expectedTokens = {565, 4923};
+    assertTokensValues(tokens, expectedTokens);
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
+        EXPECT_FALSE(rval.has_value());
+    }
+    std::string testPrompt2 = "TEST\n";
+    tokens = tokenizer->encode(testPrompt2);
+    expectedTokens = {565, 4923, 50118};
+    assertTokensValues(tokens, expectedTokens);
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
+        if (i < tokens.get_size() - 1) {
+            EXPECT_FALSE(rval.has_value());
+        } else {
+            EXPECT_TRUE(rval.has_value());
+            EXPECT_EQ(rval.value().compare(testPrompt1 + testPrompt2), 0);
+        }
+    }
+}
+
+TEST_F(TextStreamerTest, valueReturnedCacheCleared) {
+    std::string testPrompt = "TEST\n";
+    auto tokens = tokenizer->encode(testPrompt);
+    std::vector<int64_t> expectedTokens = {565, 4923, 50118};
+    assertTokensValues(tokens, expectedTokens);
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
+        if (i < tokens.get_size() - 1) {
+            EXPECT_FALSE(rval.has_value());
+        } else {
+            EXPECT_TRUE(rval.has_value());
+            EXPECT_EQ(rval.value().compare(testPrompt), 0);
+        }
+    }
+    tokens = tokenizer->encode(testPrompt);
+    for (size_t i = 0; i < tokens.get_size(); i++) {
+        std::optional<std::string> rval = this->streamer->put(static_cast<int64_t*>(tokens.data())[i]);
```

Comment:
It can probably be simplified:
```suggestion
        std::optional<std::string> rval = this->streamer->put(tokens.as<int64_t>()[i]);
```